CMakeLists.txt

```diff
@@ -466,7 +466,16 @@ else (WITH_THREADS AND Threads_FOUND)
   set (NO_THREADS 1)
 endif (WITH_THREADS AND Threads_FOUND)
 
-set (TEST_SRC_DIR \"${CMAKE_CURRENT_SOURCE_DIR}\")
+# fopen/open on Cygwin can not handle unix-type paths like /home/.... 
+# therefore we translate TEST_SRC_DIR to windows-path.
+if (CYGWIN)
+  EXECUTE_PROCESS(COMMAND cygpath.exe -m ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
Shouldn't `get_filename_component` do the job?

Reply:
I have not find a way to translate it with cmake native functions.

Reply:
You mean the command produces the same incorrect path?

Reply:
it's correct in linux terms. cmake is not aware that "/home/..." path is not good for passing into API calls on cygwin. cygwin shell translates the path before calling posix emulation layer. However if you call posix API directly in cygwin, you must use "c:/...." paths. 

Here is minimal example that demonstrates my point:
```
#include <stdio.h>
#include <sys/stat.h>
#include <fcntl.h>

int main() {
    int fd = open("/home/roman/glog/src/config.h.cmake.in", 0, O_RDONLY);
    if (fd < 0) {
        perror("could not open");
    }
    return 0;
}
```

Reply:
this fails. if you prefix it with "c:/cygwin64/" it passes. 
cmake get_filename_component  just breaks a path into components, it does not produce windows compliant paths.

Reply:
You can obtain at the absolute path using `get_filename_component(... ABSOLUTE)`. There's also `file (TO_CMAKE_PATH ...)` and to `file (TO_NATIVE_PATH ...)`.

I would rather use the native commands if possible. We can leave it as is if other options do not work.

Reply:
I've checked all these options as well as `REALPATH` before choosing execute_process. Unfortunately, none of them translate to windows native paths.